subsys/net/l2/ethernet/gptp/gptp_mi.c

```diff
@@ -960,12 +960,17 @@ static void gptp_mi_set_ps_sync_cmss(void)
 	sync_info->precise_orig_ts.second = current_time / NSEC_PER_SEC;
 	sync_info->precise_orig_ts.nanosecond = current_time % NSEC_PER_SEC;
 
-	/* TODO calculate correction field properly, rate_ratio is also set to
-	 * zero instead of being copied from global_ds as it affects the final
-	 * value of FUP correction field.
+	/* TODO calculate rate ratio and correction field properly.
+	 * Whenever time aware system is the grand master clock, we currently
+	 * make the following shortcuts:
+	 * - assuming that clock source is the local clock,
+	 *   rate_ratio is set to 1.0 instead of being copied from global_ds.
+	 * - considering that precise origin timestamp is directly inherited
+	 *   from sync egress timestamp in gptp_md_follow_up_prepare(),
+	 *   follow_up_correction_field is set to 0.
```

Comment:
Nice! Thank you for documenting these deficiencies. We might even open up a bug report for that. Looks like something that could be tagged as "[good first issue](https://github.com/zephyrproject-rtos/zephyr/labels/Good%20first%20issue)". If you like, you can do so and point to this comment. (Also not relevant for merging, just if you feel ambitious. ;-) )